servicetalk-client-api/src/main/java/io/servicetalk/client/api/DelayedRetry.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright © 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.client.api;
+
+import java.time.Duration;
+
+/**
+ * An interface that enhances any {@link Exception} to provide a constant {@link Duration delay} to be applied when
+ * it needs to be retried.
+ */
+public interface DelayedRetry {
```

Comment:
To align with `RetryableException` marker interface and highlight that it's related to exceptions, should we name it `DelayedRetryException`?

Reply:
My understanding is that it's a marker only for exceptions. @tkountis ?

Reply:
Yup, exceptions only.

Reply:
Renamed: https://github.com/apple/servicetalk/pull/2927/commits/8a1e5d435b8392c552d4c646ee47339013d43f38